components/authorize/authorize.tsx

```diff
@@ -1,41 +1,56 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
-import React from 'react';
+import React, {ReactNode} from 'react';
 import {FormattedMessage} from 'react-intl';
+import {OAuthApp} from 'mattermost-redux/types/integrations';
 
 import icon50 from 'images/icon50x50.png';
 import FormError from 'components/form_error';
 import {browserHistory} from 'utils/browser_history';
-
 import FormattedMarkdownMessage from 'components/formatted_markdown_message.jsx';
 
-export default class Authorize extends React.PureComponent {
-    static get propTypes() {
-        return {
-            location: PropTypes.object.isRequired,
-            actions: PropTypes.shape({
-                getOAuthAppInfo: PropTypes.func.isRequired,
-                allowOAuth2: PropTypes.func.isRequired,
-            }).isRequired,
-        };
-    }
+export type Params = {
+    responseType: string | null;
```

Comment:
Can these just be optional parameters instead of specifically expecting `null`?
```suggestion
    responseType?: string;
```

Reply:
I tried that initially but the issue here is that the `searchParams.get()` function on lines 68 - 72 returns `string | null`. This then throws a type error on line 75 as it's not the same type as `string | undefined`

![params](https://user-images.githubusercontent.com/46979603/106479652-d96dc600-64aa-11eb-8e20-1d41de459b40.jpg)

Reply:
Ahh, that's a problem. Okay we can leave this one alone.